### DISCUSSION TITLE
How to include DepositContract address in the chainspec?

### DISCUSSION THREAD

**Discussion by debjit-bw** [paradigmxyz/reth#12404] [Help] [answered]
In most places I see the DepositContract [address is hardcoded](https://github.com/paradigmxyz/reth/blob/9f6f63d45a1a5b9edf3087a9210e6a9170e4c1ec/crates/chainspec/src/constants.rs#L8). Is there a field name in chainspec that can set the address, block, topic in the ChainSpec? I tried `depositContract` or `deposit_contract`, none of them seem to work.

**Comment by mattsse** [comment, marked as answer]
there is, the from genesis impl does this:

https://github.com/paradigmxyz/reth/blob/f034bf23af2328516e7e763d0512ec7f34c4352f/crates/chainspec/src/spec.rs#L650-L653

and initializes the field on chainspec:

https://github.com/paradigmxyz/reth/blob/f034bf23af2328516e7e763d0512ec7f34c4352f/crates/chainspec/src/spec.rs#L660-L660

or for statics:

https://github.com/paradigmxyz/reth/blob/f034bf23af2328516e7e763d0512ec7f34c4352f/crates/chainspec/src/spec.rs#L43-L48

**Comment by debjit-bw** [comment]
that's super helpful, thanks
